source/client-side-operations-timeout/tests/tailable-awaitData.yml

```diff
@@ -245,3 +246,77 @@ tests:
               command:
                 getMore: { $$type: ["int", "long"] }
                 collection: *collectionName
+
+  - description: "apply remaining timeoutMS if less than maxAwaitTimeMS"
+    operations:
+      - name: createFindCursor
+        object: *collection
+        arguments:
+          filter: { _id:  1 }
+          cursorType: tailableAwait
+          batchSize: 1
+          maxAwaitTimeMS: 100
+          timeoutMS: 200
+        saveResultAsEntity: &tailableCursor tailableCursor
+      - name: iterateOnce
+        object: *tailableCursor
+      - name: iterateUntilDocumentOrError 
+        object: *tailableCursor
+        expectError:
+          isTimeoutError: true
+    expectEvents:
+      - client: *client
+        ignoreExtraEvents: true
+        events:
+          - commandStartedEvent:
+              commandName: find
+              databaseName: *databaseName
+          - commandStartedEvent:
+              commandName: getMore 
+              databaseName: *databaseName
+              command:
+                maxTimeMS: { $$lte: 100 }
+          - commandStartedEvent:
+              commandName: getMore 
+              databaseName: *databaseName
+              command:
+                maxTimeMS: { $$lte: 99 }
+
+  - description: "apply maxAwaitTimeMS if less than remaining timeout"
+    operations:
+      - name: failPoint
```

Comment:
According to @vbabanin this failpoint should be moved to the "apply remaining timeoutMS if less than maxAwaitTimeMS" test.